src/Schema/Directives/BuilderDirective.php

```diff
@@ -35,4 +42,22 @@ public function handleBuilder($builder, $value): object
 
         return $resolver($builder, $value, $this->definitionNode);
     }
+
+    /**
+     * Dynamically call a user-defined method to enhance the builder on a field.
+     *
+     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
+     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+     */
+    public function handleFieldBuilder(object $builder): object
+    {
+        $resolver = $this->getResolverFromArgument('method');
+        $value = $this->directiveArgValue('value');
```

Comment:
`null` might be a valid value to pass, use `directiveHasArg` to check if `value` is defined at all.